### ISSUE TITLE
[QUESTION] LoRa Creation

### ISSUE THREAD

**Issue by Flockhopper-bit** [SkyTNT/midi-model#23]
How would one go about creating a LoRa for this? Is there any plan to add a LoRa training feature within the GUI itself (for local use)?

**Comment by SkyTNT** [comment]
You need to prepare some MIDI files (ideally more than 1000). Use train.py for training, and specify the --task option as lora. The value of --data-val-split must be less than the number of your MIDI files, as this defines the number of MIDI files used for the validation set. The --gen-example-interval option determines the interval for generating samples during training, which is triggered every time a checkpoint is saved. Set this to a larger value to avoid frequent interruptions during training.

I do not plan to add LoRA training to the GUI. First, the GUI is mainly designed for inference. To support training, I would have to change the inference library from ONNXRuntime to PyTorch. However, PyTorch is very large, which is not ideal for packaging. Secondly, the training efficiency in the GUI might be relatively low.

**Comment by Epictyphlosion** [comment]
I have not tested the local version as of late, but from what I figured out the trainer can't be ran under Windows. Don't make the same mistake I did (see #7).

**Comment by SkyTNT** [comment]
> I have not tested the local version as of late, but from what I figured out the trainer can't be ran under Windows. Don't make the same mistake I did (see #7).

Apparently it works on Windows (my machine). You didn't install CUDA and PyTorch correctly and set the environment variables. Please follow the tutorials on the Internet. I have no obligation to help you solve the environment problems.

**Comment by Epictyphlosion** [comment]
> Apparently it works on Windows (my machine). You didn't install CUDA and PyTorch correctly and set the environment variables. Please follow the tutorials on the Internet. I have no obligation to help you solve the environment problems.

My apologies. I found out much of that after the fact.
